File: array and string/reverse_string.py
class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        front = 0
        rear = len(s) - 1
        
        while(front < rear):
            temp = s[front]
            s[front] = s[rear]
            s[rear] = temp
            front += 1
            rear -= 1

File: array and string/reverse_String.java
class Solution {
    public void reverseString(char[] s) {
        int front = 0;
        int rear = s.length - 1;
        
        while(front < rear)
        {
            swap(s, front, rear);
            front++;
            rear--;
        }
    }
    
    void swap(char[] s, int front, int rear)
    {
        char temp = s[front];
        s[front] = s[rear];
        s[rear] = temp;
    }
}